### PR TITLE
feat(action): hide action column when there are no visible actions

### PR DESCRIPTION
**Pull request by VMBindraban**

### CODE REVIEW

**Comment by doktordirk** [comment]
that might not be desired. eg having several tables in a row which would loose alignment now. 

ps: what the difference to showActions()?

**Comment by VMBindraban** [comment]
If you don't want to lose alignment you shouldn't use `visible`, use `disabled` instead.
I couldn't place it in showActions because of the binding.

File: src/bootstrap/datatable.html [hunk 1]
```diff
@@ -61,7 +61,7 @@
         </td>
 
         <!-- Actions -->
-        <td style="white-space: nowrap; width: 1px;" if.bind="showActions()">
+        <td style="white-space: nowrap; width: 1px;" show.bind="showActions() && hasVisibleActions">
```
Comment (doktordirk) [inline src/bootstrap/datatable.html:64]: why the change to 'show.bind'?

File: src/datatable.js [hunk 2]
```diff
@@ -137,18 +138,26 @@ export class DataTable {
 
   checkDisabled(action, row) {
     if (typeof action.disabled === 'function') {
+      this.hasVisibleActions = true;
```
Comment (doktordirk) [inline src/datatable.js]: that might get problems with the execution order of checkDisabled and checkVisibility. anyways, might not be needed, pending maybe next comment

File: src/datatable.js [hunk 3]
```diff
@@ -137,18 +138,26 @@ export class DataTable {
...
     }
 
     return false;
   }
 
   checkVisibility(action, row) {
-    if (typeof action.visible === 'function') {
-      return action.visible(row);
+    if (typeof action.visible !== 'function') {
+      return true;
```
Comment (doktordirk) [inline src/datatable.js:151]: that should set hasVisibleActions true, or? (in which case a =?: might be easier)

Comment (VMBindraban) [reply src/datatable.js:151]: I just returned early, the previous code already returned true if it wasnt a function.

File: src/bootstrap/datatable.html (hunk 1 above)
Comment (VMBindraban) [reply src/bootstrap/datatable.html:64]: `if.bind` only executes once, if 1 action is visible it must show all the `td`s.

**Review by doktordirk — APPROVED** [review]
alrighty that makes sense to me now
